### ISSUE TITLE
DojoMobileViewSceneManager must use AMD

### ISSUE THREAD

**Issue by peller** [maqetta/maqetta#3646]
(see metadata/callbacks.js in the dojo_1_8 plugin)

There are several references to Dojo objects directly from the iframe's window object.  Constructs like context.getDojo() and window.dijit should be avoided.  These should be replaced with AMD-style require statements unless somehow we can guarantee that those objects have been loaded already.  

I'm seeing intermittent failures in R&C (which may not be new) because the review Context object makes calls directly from onload which eventually end up in this class.  It's possible that we will also need to update that Context object to fire on dojo/ready so we can be sure that the loader (require) is ready to be called.

@JonFerraiolo

**Comment by peller** [comment]
@JonFerraiolo Can you take a look at the pull request and merge it if it's good?

**Comment by JonFerraiolo** [comment]
Looks good, but I have a couple of concerns:

```
+          var innerRegistry = win["require"]("dijit/registry"),
+            innerDomClass = win["require"]("dojo/dom-class"),
+            innerTopic = win["require"]("dojo/topic");
```

(Same sort of thing later on in the file in a couple of places.)

What if those Dojo classes haven't been loaded yet? I realize that in 99.9% of scenarios, these classes will be loaded by something else on the page, but is there a way to make this bulletproof?

Also, I see this check has been removed:

```
-      var dj = this.context.getDojo();
-      if(!dj){
-        return allSceneContainers;
-      }
```

I wonder if there are timing issues where `getCurrentScene()` is called before dojo has been initialized properly and the check for `dj` is necessary.

If you feel confident about these two concerns (or if you are confident that we will discover any potential problems during the M10 testing cycle), then go ahead and do the pull.

**Comment by peller** [comment]
Good points.

You're probably right that it would be more bulletproof or proper to use async requests for those three modules.  I did that in one method where it seemed reasonable to do so; for the others it was non-trivial.  I'm taking this 99% bet because these three modules are loaded by dijit/_WidgetBase

The second part, the check was removed because the Dojo reference (dojo/query) was replaced with a simple DOM QSA call.  Simpler is better.  There's a similar line of paranoid code above to check to make sure context and context.rootNode exist.  It's unclear whether any of that is necessary (I left in similar calls to check for win being defined, etc.) though I feel sometimes we'd be better off without it unless it's proven necessary.   Has the potential to hide bugs.
